---
title: "peak_analysis"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)
options(stringsAsFactors = F)
library(GenomicRanges)
library(tidyverse)
library(IRanges)
library(valr)
source("/scratch/Shares/rinn/tenaya/util/intersect_functions.R")
source("/scratch/Shares/rinn/tenaya/util/plotting_functions.R")
```

```{r}
# First we need promoters let's make them quick: (actually takes 5 min or so)
gencode_gr <- rtracklayer::import("/scratch/Shares/rinn/genomes/Homo_sapiens/Gencode/v32/gencode.v32.annotation.gtf")
# create promoters
promoters <- promoters(gencode_gr[gencode_gr$type == "gene"], upstream = 1000, downstream = 1000)
```


```{r}
#KDM1A = LSD1
#do the wget in the terminal to get the file from encode (get link from encode site, copy the download button link)
# wget https://www.encodeproject.org/files/ENCFF924FYI/@@download/ENCFF505RNR.bed.gz

kdm1a_peaks <- read_tsv("/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/data_files/KDM1A/ENCFF829FZW.bed.gz", col_names = F)
names(kdm1a_peaks) <- c('chromosome', 'start', 'end', 'name', 'score', 'strand',
                       'signalValue', 'pValue', 'qValue', 'peak_center')
write.csv(kdm1a_peaks,"/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/csv_files_primers/kdm1a_peaks_ENCFF829FZW.csv",row.names = TRUE)

# converting to Granges
kdm1a_gr <- GRanges(kdm1a_peaks$chromosome, IRanges(kdm1a_peaks$start, kdm1a_peaks$end))
# creating metaplot DF and running profile_tss
kdm1a_metaplot_df <- data.frame(x = integer(), dens = numeric(), dbp = character())
kdm1a_metaplot_df <- profile_tss(kdm1a_gr, promoters)
# plotting
ggplot(kdm1a_metaplot_df, aes(x = x, y = dens)) +
  geom_line(size = 1.5)

nrow(kdm1a_peaks)
# 31135 peaks

subsetByOverlaps(kdm1a_gr, promoters)
# 4541
```

```{r}
#PHF8
#do the wget in the terminal to get the file from encode (get link from encode site, copy the download button link)
# wget https://www.encodeproject.org/files/ENCFF924FYI/@@download/ENCFF505RNR.bed.gz

phf8_peaks <- read_tsv("/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/data_files/PHF8/ENCFF981ISM.bed.gz", col_names = F)
names(phf8_peaks) <- c('chromosome', 'start', 'end', 'name', 'score', 'strand',
                       'signalValue', 'pValue', 'qValue', 'peak_center')
write.csv(phf8_peaks,"/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/csv_files_primers/phf8_peaks_ENCFF981ISM.csv",row.names = TRUE)

# converting to Granges
phf8_gr <- GRanges(phf8_peaks$chromosome, IRanges(phf8_peaks$start, phf8_peaks$end))
# creating metaplot DF and running profile_tss
phf8_metaplot_df <- data.frame(x = integer(), dens = numeric(), dbp = character())
phf8_metaplot_df <- profile_tss(phf8_gr, promoters)
# plotting
ggplot(phf8_metaplot_df, aes(x = x, y = dens)) +
  geom_line(size = 1.5)

nrow(phf8_peaks)
# 29363

subsetByOverlaps(phf8_gr, promoters)
# 20020
```

```{r}
#DNMT1
#do the wget in the terminal to get the file from encode (get link from encode site, copy the download button link)
# wget https://www.encodeproject.org/files/ENCFF924FYI/@@download/ENCFF505RNR.bed.gz

dnmt1_peaks <- read_tsv("/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/data_files/DNMT1/ENCFF412WOK.bed.gz", col_names = F)
names(dnmt1_peaks) <- c('chromosome', 'start', 'end', 'name', 'score', 'strand',
                       'signalValue', 'pValue', 'qValue', 'peak_center')
write.csv(dnmt1_peaks,"/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/csv_files_primers/dnmt1_peaks_ENCFF412WOK.csv",row.names = TRUE)

# converting to Granges
dnmt1_gr <- GRanges(dnmt1_peaks$chromosome, IRanges(dnmt1_peaks$start, dnmt1_peaks$end))
# creating metaplot DF and running profile_tss
dnmt1_metaplot_df <- data.frame(x = integer(), dens = numeric(), dbp = character())
dnmt1_metaplot_df <- profile_tss(dnmt1_gr, promoters)
# plotting
ggplot(dnmt1_metaplot_df, aes(x = x, y = dens)) +
  geom_line(size = 1.5)

nrow(dnmt1_peaks)
# 2339

subsetByOverlaps(dnmt1_gr, promoters)
# 1061
```


```{r}
#CHD7 (named the items incorrectly)
#do the wget in the terminal to get the file from encode (get link from encode site, copy the download button link)
# wget https://www.encodeproject.org/files/ENCFF924FYI/@@download/ENCFF505RNR.bed.gz

chd1_peaks <- read_tsv("/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/data_files/CHD1/ENCFF722UJW.bed.gz", col_names = F)
names(chd1_peaks) <- c('chromosome', 'start', 'end', 'name', 'score', 'strand',
                       'signalValue', 'pValue', 'qValue', 'peak_center')
write.csv(chd1_peaks,"/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/csv_files_primers/chd1_peaks_ENCFF722UJW.csv",row.names = TRUE)

# converting to Granges
chd1_gr <- GRanges(chd1_peaks$chromosome, IRanges(chd1_peaks$start, chd1_peaks$end))
# creating metaplot DF and running profile_tss
chd1_metaplot_df <- data.frame(x = integer(), dens = numeric(), dbp = character())
chd1_metaplot_df <- profile_tss(chd1_gr, promoters)
# plotting
ggplot(chd1_metaplot_df, aes(x = x, y = dens)) +
  geom_line(size = 1.5)

nrow(chd1_peaks)
# 1177

subsetByOverlaps(chd1_gr, promoters)
# 466
```

```{r}
#CHD1 
#do the wget in the terminal to get the file from encode (get link from encode site, copy the download button link)
# wget https://www.encodeproject.org/files/ENCFF924FYI/@@download/ENCFF505RNR.bed.gz

chd11_peaks <- read_tsv("/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/data_files/CHD11/ENCFF408NUX.bed.gz", col_names = F)
names(chd11_peaks) <- c('chromosome', 'start', 'end', 'name', 'score', 'strand',
                       'signalValue', 'pValue', 'qValue', 'peak_center')
write.csv(chd11_peaks,"/scratch/Shares/rinn/Sky/rchip/analysis/00_NEW_candidate_qc/csv_files_primers/chd11_peaks_ENCFF408NUX.csv",row.names = TRUE)

# converting to Granges
chd11_gr <- GRanges(chd11_peaks$chromosome, IRanges(chd11_peaks$start, chd11_peaks$end))
# creating metaplot DF and running profile_tss
chd11_metaplot_df <- data.frame(x = integer(), dens = numeric(), dbp = character())
chd11_metaplot_df <- profile_tss(chd11_gr, promoters)
# plotting
ggplot(chd11_metaplot_df, aes(x = x, y = dens)) +
  geom_line(size = 1.5)

nrow(chd11_peaks)
# 11271

subsetByOverlaps(chd11_gr, promoters)
# 7813
```

